src/main/java/com/faforever/client/login/OAuthValuesReceiver.java

```diff
@@ -147,5 +151,19 @@ private String extractValue(String request, Pattern pattern) {
     return matcher.group(1);
   }
 
+  private void checkForError(String request) {
+    String decodedRequest = URLDecoder.decode(request, StandardCharsets.UTF_8);
```

Comment:
Instead of decoding twice I was thinking we can decode the request string once and change the regex patterns, thoughts @Sheikah45 ?

Reply:
Yes I think that would be best